### ISSUE TITLE
Any plan for WORLD vocoder?

### ISSUE THREAD

**Issue by lsq357** [r9y9/deepvoice3_pytorch#7]
Any plan for [WORLD vocoder](https://github.com/mmorise/World) for Multi-Speaker TTS

**Comment by r9y9** [comment]
Not currently planned. I wish I had more time..

**Comment by r9y9** [comment]
I will leave this open to track progress on it. Not currently planned, though.

**Comment by r9y9** [comment]
Seems like there's a folk trying to support WORLD vocoder. https://github.com/geneing/deepvoice3_pytorch

**Comment by DarkDefender** [comment]
@r9y9 Thanks for the heads up!

I'm actually really interested in how this turns out. As the WORLD vocoder is used in the "UTAU" music software. If one managed to make the network be able to train successfully with this then I think we might be able to get rid of the "sound compression" artifacts that is present in most of the current deepvoice/tacotron implementations...

And example of the sound quality possible with UTAU (and therefore WORLD):
https://www.youtube.com/watch?v=Es_5kvVtiNA

@geneing would you mind keeping us updated with your progess? Even if the results are not good.

**Comment by geneing** [comment]
Replacing Griffin-Lim with World vocoder seems to be fairly straightforward. Full transform for 22KHz signal is length 1027 vs 80 for mel output. World vocoder includes an encoder for aperiodicity and spectrogram, which reduces output to length of 131.

**Comment by lsq357** [comment]
In my view, using WORLD vocoder, the network only changes the output shape and adds multi-output,  which  WORLD vocoder need at least three parameters(f0, aperiodicity, spectrogram).
Moreover, it can add WORLD parameters(f0, aperiodicity spectrogram) and mel-outputs to loss function which speed convergence.(the idea is my guess!)

**Comment by DarkDefender** [comment]
BTW is anyone is interested in singing neural networks. Then I just found this:
http://www.dtic.upf.edu/~mblaauw/NPSS/

The spanish output sounds really awesome I think. The english and japanese sounds a little bit too stilted. But I guess that depens on what kind of dataset and music you throw at it.

**Comment by geneing** [comment]
In the view of the Tacotron 2 paper, it appears that WaveNet may be a better choice. Looking into it.

**Comment by lsq357** [comment]
It needs much more GPSs  to train Wavenet(use 32 GPUs in DeepVoice3).
I only have 2 GPUs,maybe need a few months to train.

**Comment by lsq357** [comment]
It needs much more GPUs to train Wavenet for me(in Tacotron use 32 GPUs ).
And  WORLD vocoder can use only in cpu.

**Comment by r9y9** [comment]
Does anybody have experience working on WaveNet? Is it impossible to train WaveNet with only 1 GPU in practice?

**Comment by lsq357** [comment]
I experience WaveNet on two 1080Ti GPUs, it only train 3k+ steps(asyn update) each day.,batch size =32.

I try QuasiRNN + (WaveNet)[https://github.com/tensorflow/magenta/tree/master/magenta/models/nsynth/wavenet] in DeepVoice2 or DeepVoice, but my tensorflow code of QuasiRNN not speed up! 
I only train a week, and not sucess.

**Comment by r9y9** [comment]
I started to implement the WaveNet vocoder. Check out https://github.com/r9y9/wavenet_vocoder/issues/1#issuecomment-354586299 if you are interested.

**Comment by MlWoo** [comment]
@geneing Have you trained your model with "world"? Could you provide some audio samples?

**Comment by hash2430** [comment]
I made one myself.
https://github.com/hash2430/dv3_world
Anyone who needs it are welcome to use. 
I will upload sample audios soon.
